pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,208 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer_test
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+
+	utilstest "github.com/kinvolk/inspektor-gadget/pkg/gadgets/internal/test"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+func TestCreateTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(tracer.Stop)
+}
+
+func TestStopTracerIdempotent(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func createTracer(
+	t *testing.T, config *tracer.Config, callback func(types.Event),
+) *tracer.Tracer {
+	t.Helper()
+
+	tracer, err := tracer.NewTracer(config, nil, callback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	return tracer
+}
+
+func TestTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1 (-1 because args[0] is on the first position).
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// Function to generate an event used most of the times.
+	// Returns pid of executed process.
+	generateEvent := func() (int, error) {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return 0, fmt.Errorf("running command: %w", err)
+		}
+
+		return cmd.Process.Pid, nil
+	}
+
+	tests := map[string]utilstest.Test[types.Event, int]{
+		"captures_all_events_without_configured_filters": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				return createTracer(t, &tracer.Config{}, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectAtLeastOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					Pid:       uint32(*ret),
+					Ppid:      uint32(info.Pid),
+					UID:       uint32(info.UID),
+					MountNsID: info.MountNsID,
+					Retval:    0,
+					Comm:      "cat",
+					Args:      []string{"/bin/cat", "/dev/null"},
+				}
+			}),
+		},
+		"filtering_no_event_should_be_captured": {
```

Comment:
Test says here `TestTracer/filtering_no_event_should_be_captured`, which doesn't make much sense to me. Perhaps it could still be clarified.